featurebyte/query_graph/transform/on_demand_view.py

```diff
@@ -97,7 +103,79 @@ def _post_compute(
         # it can be passed as `inputs` to the next node's post compute operation
         return var_name_or_expr
 
+    @staticmethod
+    def generate_ttl_handling_statements(
+        feature_name: str,
+        input_df_name: str,
+        output_df_name: str,
+        input_column_expr: str,
+        ttl_seconds: int,
+        var_name_generator: VariableNameGenerator,
+        comment: str = "",
+    ) -> StatementStr:
+        """
+        Generate time-to-live (TTL) handling statements for the feature or target query graph
+
+        Parameters
+        ----------
+        feature_name: str
+            Feature name
+        input_df_name: str
+            Input dataframe name
+        output_df_name: str
+            Output dataframe name
+        input_column_expr: str
+            Input column expression (to be applied for ttl handling)
+        ttl_seconds: int
+            Time-to-live (TTL) in seconds
+        var_name_generator: VariableNameGenerator
+            Variable name generator
+        comment: str
+            Comment
+
+        Returns
+        -------
+        StatementStr
+            Generated code
+        """
+        # expressions
+        subset_pit_expr = subset_frame_column_expr(
+            input_df_name, SpecialColumnName.POINT_IN_TIME.value
+        )
+        subset_feat_time_col_expr = subset_frame_column_expr(
+            input_df_name, InternalName.FEATURE_TIMESTAMP_COLUMN.value
+        )
+        subset_output_column_expr = subset_frame_column_expr(output_df_name, feature_name)
+
+        # variable names
+        req_time_var_name = var_name_generator.convert_to_variable_name(
+            variable_name_prefix="request_time", node_name=None
+        )
+        cutoff_var_name = var_name_generator.convert_to_variable_name(
+            variable_name_prefix="cutoff", node_name=None
+        )
+        ingested_var_name = var_name_generator.convert_to_variable_name(
+            variable_name_prefix="ingested_time", node_name=None
```

Comment:
I think it would be a bit more consistent to call this `feature_timestamp` since it's referred to that way in the rest of the codebase.